src/bsoncxx/include/bsoncxx/v1/document/value.hpp

```diff
@@ -29,7 +41,357 @@ namespace document {
+
+    ///
+    /// Copy assignment.
+    ///
+    /// The copied value is allocated using `operator new[]` and the deleter is set to @ref
+    /// default_deleter_type.
+    ///
+    value& operator=(value const& other) {
+        *this = value{other.view()};
+        return *this;
+    }
+
+    ///
+    /// Initialize as an empty document (or array).
+    ///
+    /// The value is equivalent to a default-initialized @ref bsoncxx::v1::document::view and the deleter is set to @ref
+    /// bsoncxx::v1::document::value::noop_deleter.
+    ///
+    /// @warning Modifying the pointed-to data after default construction is undefined behavior.
+    ///
+    value() : _data{const_cast<std::uint8_t*>(v1::document::view{}.data()), &noop_deleter} {}
+
+    ///
+    /// Initialize as owning `data` which will be freed with `deleter`.
+    ///
+    /// @par Constraints:
+    /// - `Deleter` must satisfy the requirements described by @ref deleter_type.
+    ///
+    /// @par Preconditions:
+    /// - If `data` is not null, the size of the storage region pointed to by `data` must be greater than or equal to 5.
+    /// - The "total number of bytes comprising the document" as indicated by the BSON bytes pointed-to by `data` must
+    ///   be less than or equal to the size of the storage region pointed to by `data`.
+    /// - `deleter` must be capable of freeing the storage region pointed to by `data`.
+    ///
+    template <typename Deleter, detail::enable_if_t<is_valid_deleter<Deleter>::value>* = nullptr>
+    value(std::uint8_t* data, Deleter deleter) : _data{data, std::move(deleter)} {}
+
+    ///
+    /// Initialize as owning `data` which will be freed with @ref default_deleter_type.
+    ///
+    /// @par Preconditions:
+    /// - If `data` is not null, the size of the storage region pointed to by `data` must be greater than or equal to 5.
+    /// - The "total number of bytes comprising the document" as indicated by the BSON bytes pointed-to by `data` must
+    ///   be less than or equal to the size of the storage region pointed to by `data`.
+    /// - `deleter` must be capable of freeing the storage region pointed to by `data`.
+    ///
+    explicit value(std::uint8_t* data) : value{data, default_deleter_type{}} {}
+
+    ///
+    /// Initialize as owning `ptr`.
+    ///
+    /// @par Preconditions:
+    /// - If `data` is not null, the size of the storage region pointed to by `data` must be greater than or equal to 5.
+    /// - The "total number of bytes comprising the document" as indicated by the BSON bytes pointed-to by `data` must
+    ///   be less than or equal to the size of the storage region pointed to by `data`.
+    ///
+    explicit value(unique_ptr_type ptr) : _data{std::move(ptr)} {}
+
+    ///
+    /// Initialize with a copy of the BSON bytes referenced by `view`.
+    ///
+    /// If `view` is invalid or equivalent to a default-initialized @ref bsoncxx::v1::document::view, this value is
+    /// default-initialized.
```

Comment:
Given a default-initialized `value` is now a valid empty BSON document rather than an invalid view, the "invalid -> empty" behavior for this constructor might no longer be [well-motivated](https://github.com/mongodb/mongo-cxx-driver/pull/1412#discussion_r2172492222).

Should initialization with invalid view be made to produce an invalid value?

```cpp
// Current
auto invalid = value{nullptr};
auto copy    = invalid; // value{invalid.view()} == value{view{nullptr}}

assert(copy.data() == view{}.data()); // Empty BSON document.
```

```cpp
// Better?
auto invalid = value{nullptr};
auto copy    = invalid; // value{invalid.view()} == value{view{nullptr}}

assert(copy.data() == nullptr); // Invalid.
```

Reply:
Updated so that an invalid view produces an invalid value (`this->data() == nullptr`).